cdap-gateway/src/main/java/co/cask/cdap/gateway/router/handlers/SecurityAuthenticationHttpHandler.java

```diff
@@ -171,6 +181,25 @@ private boolean validateSecuredInterception(ChannelHandlerContext ctx, HttpReque
     } else {
       AccessTokenTransformer.AccessTokenIdentifierPair accessTokenIdentifierPair =
         accessTokenTransformer.transform(accessToken);
+      AuditLogContent auditLogContent = routerAuditLookUp.getAuditLogContent(uri, httpMethod);
+      if (auditLogContent != null) {
+        List<String> headerNames = auditLogContent.getHeaderNames();
+        if (!headerNames.isEmpty()) {
+          Map<String, String> headers = new HashMap<>();
+          for (String headerName : headerNames) {
+            headers.put(headerName, msg.getHeader(headerName));
+          }
+          logEntry.setHeaders(headers);
+        }
+        if (auditLogContent.isLogRequestBody()) {
+          ChannelBuffer body = msg.getContent();
+          if (body.readable()) {
+            logEntry.setRequestBody(body.toString());
```

Comment:
Yes. That means readerIndex >= writerIndex, which means nothing we can read from the ChannelBuffer